### PR TITLE
fix(wrapper): Correctly handle directories

### LINKED ISSUE #210: Rename folders "OCP\Files\ForbiddenException: Access denied"
### Steps to reproduce
A flow has been created inside a folder that has an invisible label.
The rule is as follows:
1. If it's tagged with X rule:
2. Lock non-office files
3. Lock non-PDF files
4. Lock files other than folders
5. Block files that do not match ZIP.

Apparently everything works execpt when you try to rename a created folder.

The files can be renamed without difficulty.

### Expected behaviour
Being able to rename the folders created within the flow.

### Actual behaviour
[webdav] Fatal: Access denied

```
MOVE /remote.php/dav/files/fqsoporte/Asignaturas/Carpeta%20de%20Prueba
from xxx.xxx.xxx.xxx by fqsoporte at 2021-08-24T16:09:48-04:00
```

`{"reqId":"7Ui2ewE5EZ8Gep1uAUmF","level":4,"time":"2021-08-24T16:09:48-04:00","remoteAddr":"xxx.xxx.xxx.xxx","user":"fqsoporte","app":"webdav","method":"MOVE","url":"/remote.php/dav/files/fqsoporte/Asignaturas/Carpeta%20de%20Prueba","message":"Access denied","userAgent":"Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/92.0.4515.131 Safari/537.36 Edg/92.0.902.73","version":"22.1.0.1","exception":{"Exception":"OCP\\Files\\ForbiddenException","Message":"Access denied","Code":0,"Trace":[{"file":"/var/www/nextcloud/apps/files_accesscontrol/lib/StorageWrapper.php","line":61,"function":"checkFileAccess","class":"OCA\\FilesAccessControl\\Operation","type":"->"},{"file":"/var/www/nextcloud/apps/files_accesscontrol/lib/StorageWrapper.php","line":208,"function":"checkFileAccess","class":"OCA\\FilesAccessControl\\StorageWrapper","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/Storage/Wrapper/Wrapper.php","line":279,"function":"rename","class":"OCA\\FilesAccessControl\\StorageWrapper","type":"->"},{"file":"/var/www/nextcloud/apps/ransomware_protection/lib/StorageWrapper.php","line":275,"function":"rename","class":"OC\\Files\\Storage\\Wrapper\\Wrapper","type":"->"},{"file":"/var/www/nextcloud/lib/private/Files/View.php","line":826,"function":"rename","class":"OCA\\RansomwareProtection\\StorageWrapper","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Connector/Sabre/Node.php","line":140,"function":"rename","class":"OC\\Files\\View","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Tree.php","line":153,"function":"setName","class":"OCA\\DAV\\Connector\\Sabre\\Node","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/CorePlugin.php","line":612,"function":"move","class":"Sabre\\DAV\\Tree","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/event/lib/WildcardEmitterTrait.php","line":89,"function":"httpMove","class":"Sabre\\DAV\\CorePlugin","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":472,"function":"emit","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":253,"function":"invokeMethod","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/3rdparty/sabre/dav/lib/DAV/Server.php","line":321,"function":"start","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/lib/Server.php","line":333,"function":"exec","class":"Sabre\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/apps/dav/appinfo/v2/remote.php","line":35,"function":"exec","class":"OCA\\DAV\\Server","type":"->"},{"file":"/var/www/nextcloud/remote.php","line":166,"args":["/var/www/nextcloud/apps/dav/appinfo/v2/remote.php"],"function":"require_once"}],"File":"/var/www/nextcloud/apps/files_accesscontrol/lib/Operation.php","Line":87,"CustomMessage":"--"},"id":"6125522256785"}`

### Server configuration

**Operating system**:

**Web server:** Ubuntu 20.04

**Database:** MariaDB 10.6.3

**PHP version:** 7.4.22

**Nextcloud version:** 22.1.0

**Where did you install Nextcloud from:** tar

**Signing status:** ok

### PR DESCRIPTION
**Pull request by nickvergessen**
Fix #210

### CODE REVIEW

File: tests/Integration/features/mimetypes.feature [hunk 1]
```diff
@@ -65,3 +65,23 @@
...
+    Given User "test1" created a folder "/folder"
+    And The webdav response should have a status code "201"
+    When User "test1" moves folder "/folder" to "/folder-renamed"
+    And The webdav response should have a status code "201"
+    When User "test1" copies folder "/folder-renamed" to "/folder-copied"
+    And The webdav response should have a status code "201"
+    When User "test1" moves file "/hello" to "/hello.txt"
+    And The webdav response should have a status code "403"
+    When User "test1" copies file "/hello" to "/hello.txt"
+    And The webdav response should have a status code "403|500"
```
Comment (nickvergessen) [inline tests/Integration/features/mimetypes.feature:87]: this is a bug in server. The COPY request sends a 500 status code on Forbidden exception. seems to miss a catch somewhere.

**Review by juliushaertl — APPROVED** [review]

**Pull request merged**